app/use_cases/determine_current_roast.rb

```diff
@@ -0,0 +1,14 @@
+class DetermineCurrentRoast
+
```

Comment:
I would not use `use cases` for these types of domain concepts. A service type object like a use case is useful when there is a long list of things to do that span multiple domain objects. In this case a static method on the Roast I think is more appropriate, easier to test, and find to use in other scenarios.

You could create a static method on the Roast class called `active_roast`